Api/REST.md

```diff
@@ -1,2 +1,218 @@
 ## REST
 
+We follow RESTFUL conventions in order to have a consistency between client and server communication. We are always using JSON as standard way of communicating for making the request and getting the response.
+
+Here are some guidelines for the best practices.
+
+
+
+### Routes
+
+Standard route format, for and API that returns costumers information, should be *`api/v1/customers`*.  We achieve this by using *`Route`* attribute and .NET routing keywords such as *`{{version:apiVersion}}`*, *`[controller]`* etc.
+
+Default route should be set ether on controller level or on base controller if controllers are inheriting it.
```

Comment:
Maybe give an example, eg. 
Default route can be set on controller level :
```c#
    [ApiController]
    [Route("api/v{version:apiVersion}/customers")]
    public class CustomersController : ControllerBase
    {
...
    }
```

or specified generically on through a controller base class : ...

Mislim da mozes i taj BaseApiController stavit abstract ?